common/src/test/java/com/revenuecat/purchases/common/networking/EndpointTest.kt

```diff
@@ -82,13 +95,46 @@ class EndpointTest {
     fun `supportsSignatureValidation returns false for expected values`() {
         val expectedNotSupportsValidationEndpoints = listOf(
             Endpoint.GetAmazonReceipt("test-user-id", "test-receipt-id"),
-            Endpoint.GetOfferings("test-user-id"),
             Endpoint.PostAttributes("test-user-id"),
             Endpoint.PostDiagnostics,
-            Endpoint.GetProductEntitlementMapping
         )
         for (endpoint in expectedNotSupportsValidationEndpoints) {
             assertThat(endpoint.supportsSignatureValidation).isFalse
         }
     }
+
+    @Test
+    fun `verify needsNonceToPerformSigning is true only if supportsSignatureValidation is true`() {
+        for (endpoint in allEndpoints) {
+            if (!endpoint.supportsSignatureValidation) {
+                assertThat(endpoint.needsNonceToPerformSigning).isFalse
```

Comment:
Can you add a custom message to these? I think if one fails, it'll be hard to know which one since it's in a loop.